src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -135,6 +137,45 @@ public boolean isControlled(UUID workspaceId, UUID referenceId) {
     }
   }
 
+  public boolean updateDataReference(
+      UUID workspaceId, UUID referenceId, String name, String referenceDescription) {
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("id", referenceId.toString())
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("name", name)
+            .addValue("reference_description", referenceDescription);
+
+    StringJoiner updateStatement =
+        new StringJoiner(
+            ", ",
+            "UPDATE workspace_data_reference SET ",
+            " WHERE reference_id = :id AND workspace_id = :workspace_id");
+
+    if (name != null) {
+      updateStatement.add("name = :name");
+    }
+    if (referenceDescription != null) {
+      updateStatement.add("reference_description = :reference_description");
+    }
+
+    int rowsAffected = jdbcTemplate.update(updateStatement.toString(), params);
+    boolean updated = rowsAffected > 0;
+
+    if (updated)
```

Comment:
nit: I think this can be debug logging and can just report the updated or not flag rather than all of this.

Reply:
This is copied from the delete endpoint below, so I'm inclined to leave it as-is for now and change both in a follow-on.